src/server/utils/wallets/getSmartWallet.ts

```diff
@@ -14,13 +15,28 @@ export const getSmartWallet = async ({
   accountAddress,
 }: GetSmartWalletParams) => {
   let factoryAddress: string;
+
   try {
-    const contract = await getContract({
-      chainId,
-      contractAddress: accountAddress,
-    });
-    factoryAddress = await contract.call("factory");
+    // Note: This is a temporary solution to use cached deployed address's factory address from create-account
+    // This is needed due to a potential race condition of submitting a transaction immediately after creating an account that is not yet mined onchain
+    if (redis) {
```

Comment:
`redis` is required to run Engine now so this should never be `null` after Engine server/worker is started.